### PR TITLE
Test object listener for linked object changes

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
Just proving a theory on whether listeners are fired on collection if their linked object changes.

### CODE REVIEW

**Review by tomduncalf — COMMENTED** [review]
Just realised I had not submitted this comment!

**Comment by kraenhansen** [comment]
It deletes the vendor/.npmignore  … that seems unrelated and as a mistake?

File: integration-tests/tests/src/tests/objects.ts [hunk 1]
```diff
@@ -257,4 +258,66 @@ describe("Realm objects", () => {
       expect(persons.length).equals(1);
     });
   });
+  describe("listeners on linked objects", () => {
+    it("collection listener is called on changes to an object's linked object property", async () => {
```
Comment (kraenhansen) [inline integration-tests/tests/src/tests/objects.ts:262]:
```suggestion
    it("is called on changes to linked object", async () => {
```

File: integration-tests/tests/src/tests/objects.ts [hunk 2]
```diff
@@ -257,4 +258,66 @@ describe("Realm objects", () => {
...
+      const realm = new Realm({ schema: [Dog, Person] });
+
+      const person = realm.write(() => {
+        const tmpPerson = realm.create<IPerson>("Person", { name: "bob", age: 12 });
+        realm.create<IDog>("Dog", { age: 4, owner: tmpPerson, name: "rex" });
```
Comment (kraenhansen) [inline integration-tests/tests/src/tests/objects.ts:267]:
```suggestion
        const bob = realm.create<IPerson>("Person", { name: "bob", age: 12 });
        realm.create<IDog>("Dog", { age: 4, owner: bob, name: "rex" });
```

File: integration-tests/tests/src/tests/objects.ts [hunk 3]
```diff
@@ -257,4 +258,66 @@ describe("Realm objects", () => {
...
+        return tmpPerson;
+      });
+
+      const collection = realm.objects<IDog>("Dog");
```
Comment (kraenhansen) [inline integration-tests/tests/src/tests/objects.ts:271]:
```suggestion
      const dogs = realm.objects<IDog>("Dog");
```

File: integration-tests/tests/src/utils/perform-action-with-listener.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * @param setupListener Function called before action which can be used to setup a listener
+ */
+export const performActionWithListener = async (
+  action: () => void,
+  setupListener: (resolve: (value?: unknown) => void, reject: (reason?: any) => void) => void,
+): Promise<unknown> => {
+  return new Promise((resolve, reject) => {
+    setupListener(resolve, reject);
+    action();
+  });
```
Comment (kraenhansen) [inline integration-tests/tests/src/utils/perform-action-with-listener.ts:54]: I feel this code is a bit too small to justify the added complexity of an abstraction.

File: vendor/.npmignore [hunk 5]
```diff
@@ -1,21 +0,0 @@
-*.orig
```
Comment (kraenhansen) [inline vendor/.npmignore:1]: Looks as if this got deleted by accident.

File: integration-tests/tests/src/tests/objects.ts [hunk 6]
```diff
@@ -257,4 +258,66 @@ describe("Realm objects", () => {
       expect(persons.length).equals(1);
     });
   });
+  describe("listeners on linked objects", () => {
```
Comment (kraenhansen) [inline integration-tests/tests/src/tests/objects.ts:261]: It seems to me like you're adding listeners on collections and not on individual objects?

```suggestion
  describe("collection listeners", () => {
```

**Review by kraenhansen — CHANGES_REQUESTED** [review]
I have a few suggestions, most notably I think the `.npmignore` should be restored and the `performActionWithListener` abstraction removed.

**Comment by takameyer** [comment]
I assume this is mostly covered by now.

**Pull request closed without merging**